unit_tests/src/test_logical_paths_and_special_characters.cpp

```diff
@@ -0,0 +1,228 @@
+        REQUIRE(in_path == input.outPath);
+    }
+
+    SECTION("keep leading hat")
+    {
+        const std::string in_path = "/zone/^data_object";
+        std::strncpy(input.inPath, in_path.data(), in_path.size());
+
+        REQUIRE(parseRodsPath(&input, &env) == 0);
+        REQUIRE(in_path == input.outPath);
+    }
+
+    SECTION("keep trailing tilde")
+    {
+        const std::string in_path = "/zone/data_object~";
+        std::strncpy(input.inPath, in_path.data(), in_path.size());
+
+        REQUIRE(parseRodsPath(&input, &env) == 0);
+        REQUIRE(in_path == input.outPath);
+    }
+
+    SECTION("keep trailing period")
+    {
+        const std::string in_path = "/zone/data_object.";
+        std::strncpy(input.inPath, in_path.data(), in_path.size());
+
+        REQUIRE(parseRodsPath(&input, &env) == 0);
+        REQUIRE(in_path == input.outPath);
+    }
+
+    SECTION("keep trailing hat")
+    {
+        const std::string in_path = "/zone/data_object^";
+        std::strncpy(input.inPath, in_path.data(), in_path.size());
+
+        REQUIRE(parseRodsPath(&input, &env) == 0);
+        REQUIRE(in_path == input.outPath);
+    }
+}
+
+TEST_CASE("test getLastPathElement")
+{
+    using namespace std::string_literals;
+
+    char last_path_element[MAX_NAME_LEN]{};
+
+    SECTION("/path/to/.")
+    {
+        std::string path = "/path/to/.";
+        REQUIRE(getLastPathElement(&path[0], last_path_element) == 0);
+        REQUIRE(last_path_element == ""s);
+    }
+
+    SECTION("/path/to/..")
+    {
+        std::string path = "/path/to/..";
+        REQUIRE(getLastPathElement(&path[0], last_path_element) == 0);
+        REQUIRE(last_path_element == ""s);
+    }
+
+    SECTION("/path/to/foo")
+    {
+        std::string path = "/path/to/foo";
+        REQUIRE(getLastPathElement(&path[0], last_path_element) == 0);
+        REQUIRE(last_path_element == "foo"s);
+    }
+
+    SECTION("/path/to/foo/.")
+    {
+        std::string path = "/path/to/foo/.";
+        REQUIRE(getLastPathElement(&path[0], last_path_element) == 0);
+        REQUIRE(last_path_element == ""s);
+    }
+
+    SECTION("/path/to/foo/..")
+    {
+        std::string path = "/path/to/foo/..";
+        REQUIRE(getLastPathElement(&path[0], last_path_element) == 0);
+        REQUIRE(last_path_element == ""s);
+    }
+
+    SECTION("/path/to/foo/")
+    {
+        std::string path = "/path/to/foo/";
+        REQUIRE(getLastPathElement(&path[0], last_path_element) == 0);
+        REQUIRE(last_path_element == "/path/to/foo"s);
+    }
+
+    SECTION("/path/to/foo.")
+    {
+        std::string path = "/path/to/foo.";
+        REQUIRE(getLastPathElement(&path[0], last_path_element) == 0);
+        REQUIRE(last_path_element == "foo."s);
+    }
+
+    SECTION("/path/to/foo~")
+    {
+        std::string path = "/path/to/foo~";
+        REQUIRE(getLastPathElement(&path[0], last_path_element) == 0);
+        REQUIRE(last_path_element == "foo~"s);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [String literal compared with variable 'last_path_element'. Did you intend to use strcmp() instead?](https://app.codacy.com/app/trel/irods/pullRequest?prid=4593032)